mp/src/vgui2/vgui_controls/CvarTextEntry.cpp

```diff
@@ -62,6 +62,25 @@ void CvarTextEntry::GetSettings(KeyValues* pOutResource)
     pOutResource->SetString("cvar_name", m_cvarRef.GetName());
 }
 
+void CvarTextEntry::SetText(const char* text)
+{
+    if (!text || !text[0]) 
+        return;
+
+    if (text[strnlen(text, MAX_CVAR_TEXT) - 1] == '.')
+    {
+        // trying to enter a decimal number so set the text normally
+        BaseClass::SetText(text);
+    }
+    else // not in the middle of setting a decimal number
+    {
+        // making sure the formatting is correct (removing trailing zeros via %g)
+        char newText[MAX_CVAR_TEXT];
+        Q_snprintf(newText, MAX_CVAR_TEXT, "%g", atof(text));
```

Comment:
Wait, new question: what happens if this text entry isn't numeric only? If I set the string for like my name or something using this, what happens when the string isn't a number?

Reply:
I tested it and no matter the input, it sets the text entry to whatever float value pops out of the `atof(text)` call, which the class then recognizes isn't equal to the cvar's actual value & resets it. Specifically, `MaxBeenmodifiedExternally()` returns true in `OnThink()` and the textentry is reset.

When the textentry is being set to the cvar's actual value, which is done in the `Reset()` function, `MaxBeenmodifiedExternally()` returns false.

Reply:
Right I mean like, if we have a `CvarTextEntry` for a name or file path, wouldn't you want to set it as just the string? My point is, it's bad to assume this is only ever going to be numeric, and plus there's already a `IsNumericInputOnly` bool.

I'd probably put a
```
if (IsNumericInputOnly())
{
    <this code>
}
else
{
    BaseClass::SetText(text);
}
```

branch in here.

Reply:
Ah I didn't realize this only accounted for `CvarTextEntry`s that were numeric. Added this check.